rasa_nlu/tokenizers/whitespace_tokenizer.py

```diff
@@ -31,7 +32,11 @@ def process(self, message, **kwargs):
     def tokenize(self, text):
         # type: (Text) -> List[Token]
 
-        words = text.split()
+        # replace punctuation with space to preserve character indices
+        # there is space after punctuation
+        # because we do not want to replace 10.000 with 10 000
+        words = re.sub(r'[.,!?]\s', '  ', text + ' ')[:-1].split()
```

Comment:
how about `r'[.,!?](\s|$)', '  ', text`

should it maybe even be 

`r'[.,!?]+(\s|$)', '  ', text` to catch something like `hey there...`

Reply:
of course, index of token will be still the same because we look in original text. Will change that